src/antsibull/announcements.py

```diff
@@ -0,0 +1,305 @@
+) -> None:
+    subject = SUBJECT.format(info.template_vars["version"])
+    body = get_body(directory, "ansible-email-announcement.txt")
+    params_dict = FORUM_PARAMS | dict(title=subject, body=body)
+    params = "?" + urlencode(params_dict, quote_via=url_quote)
+    url = urljoin(ANSIBLE_FORUM_URL, "new-topic") + params
+    webbrowser.open(url)
+
+
+def email_announcement_webbrowser(
+    directory: Path,
+    info: AnnouncementsInfo,
+    ctx: SendCtx,  # pylint: disable=unused-argument
+) -> None:
+    subject = SUBJECT.format(info.template_vars["version"])
+    body = get_body(directory, "ansible-email-announcement.txt")
+    params_dict: dict[str, str] = {"subject": subject, "body": body}
+    params = "?" + urlencode(params_dict, quote_via=url_quote)
+    url = "mailto:" + ",".join(MAIL_RECIPIENTS) + params
+    webbrowser.open(url)
+
+
+def matrix_announcement(
+    directory: Path,
+    info: AnnouncementsInfo,  # pylint: disable=unused-argument
+    ctx: SendCtx,
+) -> None:
+    body = get_body(directory, "ansible-matrix-announcement.md")
+    try:
+        forum_url = input("Enter the URL to the forum post: ")
+    # Ctrl+D
+    except EOFError:
+        print("Continuing...")
+        return
+    body = body.replace("<FORUM LINK>", forum_url)
+    message = dedent(
+        """
+    Please open your Matrix client and send the message to:
+    - #community:ansible.com
+    - #packaging:ansible.com
+    - #social:ansible.com (mention @newsbot)
+    """
+    )
+    if ctx.clipboard:
+        pyperclip.copy(body)
+        message += "\nThe message has been copied to your clipboard"
+    else:
+        message += "---\n" + body
+    print(message)
+
+
+ACTIONS: dict[str, Callable[[Path, AnnouncementsInfo, SendCtx], None]] = {
+    "forum": forum_announcement_webbrowser,
+    "email": email_announcement_webbrowser,
+    "matrix": matrix_announcement,
+}
+
+
+@dataclass
+class SendCtx:
+    # Whether actions can write to the clipboard
+    clipboard: bool
+
+
+def send_announcements_command() -> int:
+    """
+    Interactively send announcements created by the `announcements` subcommand.
+    """
+    app_ctx = app_context.app_ctx.get()
+    announcements_dir: Path = app_ctx.extra["announcements_dir"]
+    actions: set[str] = set(app_ctx.extra["send_actions"])
+    clipboard: bool = app_ctx.extra["clipboard"]
+
+    return _send_announcements_command(announcements_dir, actions, clipboard)
+
+
+def _send_announcements_command(
+    announcements_dir: Path, actions: set[str], clipboard: bool
+) -> int:
+    info = load_announcements_json(announcements_dir / "announcements.json")
+    send_ctx = SendCtx(clipboard)
+    # Loop over ACTIONS this way to make sure that order is preserved.
+    # For example, we want the Matrix announcement to happen last
+    for action in ACTIONS:
+        if action not in actions:
+            continue
+        print(f"Handling {action}...")
+        ACTIONS[action](announcements_dir, info, send_ctx)
```

Comment:
Why not do this in async manner and start all the announcement processes concurrently?

Reply:
This is sync on purpose. One of the actions requires user input, and I don't really see much value in making them async to begin with. It's pretty much just running subprocesses to open browser tabs.